packages/react/src/hooks/utils.ts

```diff
@@ -0,0 +1,34 @@
+import IsomorphicClerk from '../isomorphicClerk';
+
+/**
+ * @internal
+ */
+const clerkLoaded = (isomorphicClerk: IsomorphicClerk) => {
+  return new Promise<void>(resolve => {
+    if (isomorphicClerk.loaded) {
+      resolve();
+    }
+    isomorphicClerk.addOnLoaded(resolve);
+  });
+};
+
+/**
+ * @internal
+ */
+export const createGetToken = (isomorphicClerk: IsomorphicClerk) => async (options: any) => {
+  await clerkLoaded(isomorphicClerk);
```

Comment:
❓Can you please elaborate why such utils are required for `useAuth` since `<ClerkLoaded/>` is used? Is this for SSR?

Reply:
I'm not exactly sure what you mean by `ClerkLoaded`. Although `ClerkLoaded` can be used, it's certainly not required for `useAuth` to work.

For SSR-enabled apps, the userId and sessionId are always populated. However, during CSR and before clerkjs has loaded, calling signOut or getToken wouldn't behave as expected since the actual behavior is on window.Clerk and not IsomorphicClerk.

This particular hook returns values that can either be initialised during SSR or can be awaited (like the 2 methods). I introduced this `await clerkLoaded` internal helper to allow calling signOut/getToken without the need for isLoaded: both these methods transparently await for clerkjs to load before resolving 

This functionality can be moved to isomorphicClerk in the future and be used in other apis as well

Reply:
Thanks for explaining